UefiCpuPkg/PiSmmCpuDxeSmm/NonMmramMapStandaloneMm.c

```diff
@@ -158,15 +172,19 @@ BuildMemoryMapFromResDescHobs (
   Hob.Raw = GetFirstHob (EFI_HOB_TYPE_RESOURCE_DESCRIPTOR);
   while (Hob.Raw != NULL) {
     if ((Hob.ResourceDescriptor->ResourceAttribute & MM_RESOURCE_ATTRIBUTE_LOGGING) == 0) {
-      ASSERT (Index < Count);
-      (*MemoryRegion)[Index].Base      = Hob.ResourceDescriptor->PhysicalStart;
-      (*MemoryRegion)[Index].Length    = Hob.ResourceDescriptor->ResourceLength;
-      (*MemoryRegion)[Index].Attribute = EFI_MEMORY_XP;
-      if (Hob.ResourceDescriptor->ResourceAttribute == EFI_RESOURCE_ATTRIBUTE_READ_ONLY_PROTECTED) {
-        (*MemoryRegion)[Index].Attribute |= EFI_MEMORY_RO;
+      ResourceHobEnd = Hob.ResourceDescriptor->PhysicalStart + Hob.ResourceDescriptor->ResourceLength;

```

Comment:
don't need change here.

Reply:
You mean remove line 175 directly use the variable in line 177? I'm fine with it

Reply:
i mean once we assert, those code are not needed.